From b30ef98044b61c069cf12b502d6fb13a50e63936 Mon Sep 17 00:00:00 2001
From: Evangelos Ribeiro Tzaras <devrtz@fortysixandtwo.eu>
Date: Wed, 21 Sep 2022 08:30:52 +0200
Subject: [PATCH 2/3] log: Guard against setting g_log_set_writer_func()
 multiple times

While it was always invalid it is enforced as of glib 2.74.

This fixes tests failing like:

GLib-ERROR **: 08:25:34.271: g_log_set_writer_func() called multiple times

Closes: https://gitlab.gnome.org/World/Phosh/phosh/-/issues/841
Part-of: <https://gitlab.gnome.org/World/Phosh/phosh/-/merge_requests/1148>
---
 src/log.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/log.c b/src/log.c
index 8e6770b2..c2372e5a 100644
--- a/src/log.c
+++ b/src/log.c
@@ -18,7 +18,7 @@
 /* these are filtered by G_MESSAGES_DEBUG by the default log handler */
 #define INFO_LEVELS (G_LOG_LEVEL_INFO | G_LOG_LEVEL_DEBUG)
 
-
+static gboolean       _log_writer_func_set;
 static char          *_log_domains;
 G_LOCK_DEFINE_STATIC (_log_domains);
 
@@ -132,6 +132,10 @@ phosh_log_set_log_domains (const char *domains)
   _log_domains = g_strdup (domains);
   G_UNLOCK (_log_domains);
 
+  if (_log_writer_func_set)
+    return;
+
   g_log_set_writer_func ((GLogWriterFunc)phosh_log_writer_default,
                          NULL, NULL);
+  _log_writer_func_set = TRUE;
 }
-- 
2.37.2

